src/components/settings/apply-license/ApplyLicenseForm.jsx

```diff
@@ -3,32 +3,41 @@ import { Button, Input, Form, Card } from 'antd';
 import FormItemLabel from "../../form-item-label/FormItemLabel";
 import { useEffect } from 'react';
 
-const ApplyLicenseForm = ({ clusterName, handleSubmit }) => {
+const ApplyLicenseForm = ({ clusterName, handleSubmit, handleSubmitOfflineLicense, licenseMode }) => {
   const [form] = Form.useForm()
 
   useEffect(() => {
     form.setFieldsValue({ clusterName })
   }, [clusterName])
 
   const handleSubmitClick = ({ clusterName, licenseKey, licenseValue }) => handleSubmit(clusterName, licenseKey, licenseValue)
+  const handleOfflineLicenseClick = ({licenseKey}) => handleSubmitOfflineLicense(licenseKey)
   return (
     <Card>
       <Form
         form={form}
-        onFinish={handleSubmitClick}
+        onFinish={licenseMode === 'offline' ? handleOfflineLicenseClick : handleSubmitClick}
         initialValues={{ clusterName }}
       >
-        <FormItemLabel name="Provide cluster name" description="Cluster name is used for you to identify the cluster associated with a license key" />
-        <Form.Item name="clusterName" rules={[{ required: true, message: "Please provide a cluster name!" }]}>
-          <Input placeholder="Cluster name" />
-        </Form.Item>
-        <FormItemLabel name="License key details" />
-        <Form.Item name="licenseKey" rules={[{ required: true, message: "Please provide a license key!" }]}>
-          <Input placeholder="License key" addonBefore="Key" />
-        </Form.Item>
-        <Form.Item name="licenseValue" rules={[{ required: true, message: "Please provide a license key secret!" }]}>
-          <Input.Password placeholder="License key secret" addonBefore="Secret" />
-        </Form.Item>
+        {licenseMode === 'online' && <React.Fragment>
```

Comment:
Let's show the online form when licenseMode !== "offline". This way even if in some case, space cloud doesn't send licenseMode value, we still would be showing at least the inline form.